element_array_ephys/probe.py

```diff
@@ -259,3 +204,63 @@ class Electrode(dj.Part):
         -> master
         -> ProbeType.Electrode
         """
+
+
+def build_electrode_layouts(
+    probe_type: str,
+    site_count_per_shank: int,
+    col_spacing: float = None,
+    row_spacing: float = None,
+    white_spacing: float = None,
+    col_count_per_shank: int = 1,
+    shank_count: int = 1,
+    shank_spacing: float = None,
+    y_origin="bottom",
+) -> list[dict]:
+
+    """Builds electrode layouts.
+
+    Args:
+        probe_type (str): probe type (e.g., "neuropixels 1.0 - 3A")
+        site_count_per_shank (int): site count per shank.
+        col_spacing (float): (um) horizontal spacing between sites. Defaults to None (single column).
+        row_spacing (float): (um) vertical spacing between columns. Defaults to None (single row).
+        white_spacing (float): (um) offset spacing. Defaults to None.
+        col_count_per_shank (int): number of column per shank. Defaults to 1 (single column).
+        shank_count (int): number of shank. Defaults to 1 (single shank).
+        shank_spacing (float): (um) spacing between shanks. Defaults to None (single shank).
+        y_origin (str): {"bottom", "top"}. y value decrements if "top". Defaults to "bottom".
+    """
+    row_count = int(site_count_per_shank / col_count_per_shank)
+    x_coords = np.tile(
+        np.arange(0, (col_spacing or 1) * col_count_per_shank, (col_spacing or 1)),
+        row_count,
+    )
+    y_coords = np.repeat(np.arange(row_count) * (row_spacing or 1), col_count_per_shank)
```

Comment:
What motivated the use of `None` defaults here? Would it not have the same result with `1` as default and no `or` operators here?

If the following suggestion is accepted, please also change the `def` and docstring

```suggestion
        np.arange(0, col_spacing * col_count_per_shank, col_spacing),
        row_count,
    )
    y_coords = np.repeat(np.arange(row_count) * row_spacing, col_count_per_shank)
```

Reply:
While either scheme will yield the same results, I've purposefully set the default to `None` here because users may be misled to think there's an actual value associated with those (e.g., 1 mm spacing ?) when there isn't.

Reply:
Makes sense. Thanks for your response